runtime-sdk/src/modules/core/mod.rs

```diff
@@ -538,7 +558,6 @@ impl<Cfg: Config> Module<Cfg> {
         let bs_max_iters = Self::estimate_gas_search_max_iters(ctx);
 
         // Update the address used within the transaction when caller address is passed.
-        let mut extra_gas = 0;
         if let Some(caller) = args.caller.clone() {
             let address_spec = transaction::AddressSpec::Internal(caller);
             match args.tx.auth_info.signer_info.first_mut() {
```

Comment:
we don't entirely wipe the signer info, we only replace the first one

Reply:
Good point, although the (currently deployed) runtimes don't currently allow any additional signers we should still just get rid of those in case the caller is overriden.

Reply:
Fixed.